packages/react-server/src/ReactFlightServer.js

```diff
@@ -553,6 +553,10 @@ function serializeUndefined(): string {
   return '$undefined';
 }
 
+function serializeBigInt(n: bigint): string {
+  return '$n' + n.toString(16);
```

Comment:
We usually use base 16 just to make the ids a little shorter. They're pretty readable regardless when they don't represent something other than the same id.

However, for big int, that's usually a human readable value. It might be worse the longer form base 10 just to make it more debuggable.

It's also how you'd write it when hard coded in code. The purpose of using the `n` is so it looks like how you'd write in code. Similar to the reason to use slashes for reg exp. Makes it a little more debuggable. Since the `n` format is base 10, it makes sense to encode it in base 10 here too.